yastd/dbus_api.md

```diff
@@ -0,0 +1,29 @@
+
+## get the list of available languages
+
+Iface: o.o.YaST.Installer1.Language
+
+stupid api:
+  AvailableLanguages -> array(string)
+
+minimal working api:
+  needed: human readable name (how localized?), identifier
+
+identifiers: maybe LanguageTag https://www.rubydoc.info/github/yast/yast-packager/master/LanguageTag
+- move it to yast-yast2
+- link to the standard from yard 
+- see https://tools.ietf.org/html/rfc4647 Matching of Language Tags
+- see https://lists.opensuse.org/archives/list/yast-devel@lists.opensuse.org/message/D52PSZ7TRID2RVM6CE6K2C2RUNNGOS6Z/
```

Comment:
well dbus api design document suggest using enums. What about having just simple plain enums here?

Reply:
Enums... This example will illustrate why I dislike enums on D-Bus:
```console
$ for p in Capabilities Ip4Connectivity Ip6Connectivity Metered State; do 
  printf "$p: "; \
  busctl --system get-property org.freedesktop.NetworkManager \
    /org/freedesktop/NetworkManager/Devices/3 \
    org.freedesktop.NetworkManager.Device $p; \
  done
Capabilities: u 1
Ip4Connectivity: u 4
Ip6Connectivity: u 3
Metered: u 4
State: u 100
```

I mean, while D-Bus has _many_ provisions to make interfaces _discoverable_, using enums is a sure way to kill that discoverability. If we use enums I will argue for adding an interface to expose the number-name mapping.

Anyway, I digress. Enjms are surely unfit for languages/locales, which are objects (simple objects, or complex strings), not members of an enumeration.